scripts/tools/TaskQueueMaintenance.php

```diff
@@ -0,0 +1,248 @@
+<?php
+
+declare(strict_types=1);
+
+namespace oat\taoTaskQueue\scripts\tools;
+
+use common_persistence_Manager;
+use Laminas\ServiceManager\ServiceLocatorAwareInterface;
+use Laminas\ServiceManager\ServiceLocatorAwareTrait;
+use oat\oatbox\action\Action;
+use oat\oatbox\reporting\Report;
+use oat\tao\model\taskQueue\TaskLogInterface;
+use oat\tao\model\taskQueue\TaskLog\TaskLogFilter;
+use oat\tao\model\taskQueue\TaskLog\Broker\TaskLogBrokerInterface;
+
+
+class TaskQueueMaintenance implements Action, ServiceLocatorAwareInterface
+{
+    use ServiceLocatorAwareTrait;
```

Comment:
please use the contructor Dependency Injection

Reply:
I investigated the possibility of using DI via constructor and looks like in case of a CLI script it would require service registration as constructor arguments are not passed automatically in case of cli scripts.
I can implement DI in another way - by adding an init method, something like:
private function initDependencies(): void
{
    if ($this->taskLog === null) {
        $this->taskLog = $this->getServiceLocator()->get(TaskLogInterface::SERVICE_ID);
    }
}
Let me know if I should do this or we can just keep the existing approach.

Reply:
I tried to do it:
1. With constructor it would require the service/factory registration and one additional config file as in CLI scripts arguments form constructor are not passed automatically.
2. With lazy initDependencies method the code looks messy as for me :) 
Maybe we can discuss it in more details what would be the best approach in this case?